delta-lake/common/src/main/databricks/scala/com/nvidia/spark/rapids/delta/DatabricksDeltaProvider.scala

```diff
@@ -172,6 +182,116 @@ object DatabricksDeltaProvider extends DeltaProviderImplBase {
+          .map(_.getBoolean(outerObj))
+        val options = getField(outerClass,
+          "com$databricks$sql$transaction$tahoe$catalog$WriteIntoDeltaBuilder$$options").map { f =>
+          f.get(outerObj).asInstanceOf[mutable.HashMap[String, String]]
+        }
+        if (forceOverwrite.isDefined && options.isDefined) {
+          Some(DeltaWriteV1Config(deltaLog, forceOverwrite.get, options.get))
+        } else {
+          meta.willNotWorkOnGpu(s"write class has unsupported outer class $outerClass")
+          None
+        }
+      } else {
+        meta.willNotWorkOnGpu(s"write class has unsupported outer class $outerClass")
+        None
+      }
+    }.getOrElse {
+      meta.willNotWorkOnGpu(s"write class $writeClass is not supported")
+      None
+    }
+  }
+
+  override def tagForGpu(
+      cpuExec: AppendDataExecV1,
+      meta: AppendDataExecV1Meta): Unit = {
+    if (!meta.conf.isDeltaWriteEnabled) {
+      meta.willNotWorkOnGpu("Delta Lake output acceleration has been disabled. To enable set " +
+        s"${RapidsConf.ENABLE_DELTA_WRITE} to true")
+    }
+    val deltaTable = cpuExec.table.asInstanceOf[DeltaTableV2]
+    val tablePath = if (deltaTable.catalogTable.isDefined) {
+      new Path(deltaTable.catalogTable.get.location)
+    } else {
+      DeltaDataSource.parsePathIdentifier(cpuExec.session, deltaTable.path.toString,
+        deltaTable.options)._1
+    }
+    val deltaLog = DeltaLog.forTable(cpuExec.session, tablePath, deltaTable.options)
+    RapidsDeltaUtils.tagForDeltaWrite(meta, cpuExec.plan.schema, Some(deltaLog),
+      deltaTable.options, cpuExec.session)
+    extractWriteV1Config(meta, deltaLog, cpuExec.write).foreach { writeConfig =>
+      meta.setCustomTaggingData(writeConfig)
+    }
+  }
+
+  override def convertToGpu(
+      cpuExec: AppendDataExecV1,
+      meta: AppendDataExecV1Meta): GpuExec = {
+    val writeConfig = meta.getCustomTaggingData match {
+      case Some(c: DeltaWriteV1Config) => c
+      case _ => throw new IllegalStateException("Missing Delta write config from tagging pass")
+    }
+    val deltaLog = writeConfig.deltaLog
+    val gpuWrite = new V1Write {
+      override def toInsertableRelation(): InsertableRelation = {
+        new InsertableRelation {
+          override def insert(data: DataFrame, overwrite: Boolean): Unit = {
+            val session = data.sparkSession
+
+            // TODO: Get the config from WriteIntoDelta's txn.
+            val cpuWrite = WriteIntoDelta(
+              deltaLog,
+              if (writeConfig.forceOverwrite) SaveMode.Overwrite else SaveMode.Append,
+              new DeltaOptions(writeConfig.options.toMap, session.sessionState.conf),
+              Nil,
+              DeltaLogShim.getMetadata(deltaLog).configuration,
+              data)
+            val gpuWrite = GpuWriteIntoDelta(new GpuDeltaLog(deltaLog, meta.conf), cpuWrite)
+            gpuWrite.run(session)
+
+            // TODO: Push this to Apache Spark
```

Comment:
Should we file an issue for this?

Reply:
This is from the original Delta source at https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala#L404-L406

Reply:
:(